lib/combinat.gi

```diff
@@ -2446,6 +2446,72 @@ InstallGlobalFunction( "IteratorOfPartitions", function( n )
     end );
 
 
+#############################################################################
+##
+#F  IteratorOfPartitionsSet( <set> )
+##
+##  all unordered partitions of the set <A>set</A> into pairwise disjoint
+##  nonempty sets.
+##
+InstallGlobalFunction( IteratorOfPartitionsSet, function( s )
+    local nextIterator, isDone, m, out, i, shallowCopy;
+
+      if not IsSet(s) then
+          Error( "<s> must be a set" );
+      fi;
+
+    nextIterator := function(iter)
```

Comment:
Could you please insert a comment which explains what this algorithm is? E.g. if it has a name, give that; if there is a reference, give that?

I know that a lot of existing code doesn't that kind of thing either -- but I think that's a major issue.

Reply:
I have added a description. Is this satisfactory?

Reply:
Thanks, just the mention of "restricted growth strings" is super helpful, as that allowed me to find papers discussing it, and also find it in TAOCP Volume 4A by Knuth.

I've taken the liberty of squashing your commits and fixing a few more formatting issues and typos. I also change `s := s` to `s := Immutable(s)`, to be resilient against changes to the input set during the iteration.

Reply:
Awesome, cheers!